### DISCUSSION TITLE
对于Fish Shell的支持。

### DISCUSSION THREAD

**Discussion by moqsien** [gvcgo/version-manager#35] [Ideas]
如题。

**Comment by zzhaolei** [comment]
有些关于 env 和 shell 的问题就先在这里讨论吧。

周末我看了一下现有的对 env 和 path 的实现，发现了一些问题。

1. 可以添加多个同名的环境变量
2. 删除环境变量时可能会将同名的 path 也一起删除（比如 `export PATH=/test:$PATH`，当我删除 test 的环境变量时，这个 path 也会被删除）
3. 添加 path 或环境变量后，需要重新 source .zshrc 才能生效

基于上面的一些问题，我建议删除 env 选项，由用户自己去管理环境变量。（可以提供 vm_env 的文件，但让用户自己决定在哪里添加）

**Reply by moqsien** [reply]
也可以。这个功能不是必须的。

**Reply by zzhaolei** [reply]
那就只剩 hook、添加 vmr 和 bin path 到环境 path 的功能了，我有空了改下

**Comment by moqsien** [comment]
> 那就只剩 hook、添加 vmr 和 bin path 到环境 path 的功能了，我有空了改下

gvcgo社区下面有一个好用的免脚本多平台编译go项目的工具，可以查看[gobuilder](https://github.com/gvcgo/gobuilder)。

可以go install一键安装，然后使用**gber b**就可以在vmr项目的任何目录下编译该项目了。

当然gobuilder是通用的，其他项目也可以一键跨平台免脚本编译。

**Comment by moqsien** [comment]
已实现。
